src/bentoml/_internal/cloud/deployment.py

```diff
@@ -1,73 +1,260 @@
 from .schemas import DeploymentListSchema
+from .schemas import DeploymentTargetType
 from .schemas import CreateDeploymentSchema
+from .schemas import DeploymentTargetConfig
+from .schemas import DeploymentTargetHPAConf
+from .schemas import DeploymentTargetCanaryRule
+from .schemas import DeploymentTargetRunnerConfig
 from ...exceptions import BentoMLException
 
+logger = logging.getLogger(__name__)
+
+
+@attr.define
+class Resource:
+    @classmethod
+    def for_hpa_conf(cls, **kwargs) -> DeploymentTargetHPAConf:
+        return bentoml_cattr.structure(kwargs, DeploymentTargetHPAConf)
+
+    @classmethod
+    def for_runner(cls, **kwargs) -> DeploymentTargetRunnerConfig:
+        exclusive_api_server_key = {
+            v for v in kwargs if v not in attr.fields_dict(DeploymentTargetRunnerConfig)
+        }
+        return bentoml_cattr.structure(
+            {k: v for k, v in kwargs.items() if k not in exclusive_api_server_key},
+            DeploymentTargetRunnerConfig,
+        )
+
+    @classmethod
+    def for_api_server(cls, **kwargs) -> DeploymentTargetConfig:
+        return bentoml_cattr.structure(kwargs, DeploymentTargetConfig)
+
 
-class deployment:
-    def list_deployment(
-        self, context: str | None = None, clusterName: str | None = None
+class Deployment:
+    @classmethod
+    def _create_deployment(
+        cls,
+        context: str | None = None,
+        cluster_name: str | None = None,
+        *,
+        create_deployment_schema: CreateDeploymentSchema,
+    ) -> DeploymentSchema:
+
+        yatai_rest_client = get_rest_api_client(context)
+        if cluster_name is None:
+            cluster_name = default_context_name
+        for target in create_deployment_schema.targets:
+            res = yatai_rest_client.get_bento(target.bento_repository, target.bento)
+            if res is None:
+                raise BentoMLException(
+                    f"Create deployment: {target.bento_repository}:{target.bento} does not exist"
+                )
+        res = yatai_rest_client.get_deployment(
+            cluster_name,
+            create_deployment_schema.kube_namespace,
+            create_deployment_schema.name,
+        )
+        if res is not None:
+            raise BentoMLException("Create deployment: Deployment already exists")
+        res = yatai_rest_client.create_deployment(
+            cluster_name, schema_to_json(create_deployment_schema)
+        )
+        if res is None:
+            raise BentoMLException("Create deployment request failed")
+        return res
+
+    @classmethod
+    def create(
+        cls,
+        deployment_name: str,
+        bento_repository: Tag | str,
+        description: str = None,
+        cluster_name: str = default_context_name,
+        kube_namespace: str = default_kube_namespace,
+        resource_instance: str | None = None,
+        cpu: str | None = None,
+        gpu: str | None = None,
+        memory: str | None = None,
```

Comment:
from the discussion yesterday, it seems like we want to hid this away from user?